test/induction.lean

```diff
@@ -528,6 +528,21 @@ end
 
 end rose
 
+-- When using `cases' t` or `induction' t`, where `t` is a complex term,
+-- the context must be generalised over `t`.
+example {n m : ℕ} (h : n + 1 = m) : n + 1 = m :=
+begin
+  cases' n + 1,
+  { exact h },
+  { exact h }
+end
+
+example {n m : ℕ} (h : n + 1 = m) : n + 1 = m :=
+begin
+  induction' n + 1,
+  { exact h },
+  { exact h }
+end
```

Comment:
Is there a `guard` tactic you can add to these tests to verify the goals are what they should be?

Reply:
Sure. I've pushed a commit to that effect.